hibernate-core/src/main/java/org/hibernate/engine/internal/Cascade.java

```diff
@@ -196,6 +202,30 @@ else if ( action.performOnLazyProperty() && type.isEntityType() ) {
 		}
 	}
 
+	private static void registerUninitializedLazyCollectionEntries(EventSource eventSource, EntityPersister persister, Object parent) {
+		if ( !persister.hasCollections() || !persister.hasUninitializedLazyProperties( parent ) ) {
```

Comment:
what about checking `persister.getBytecodeEnhancementMetadata().isEnhancedForLazyLoading()`?

Reply:
That's the first thing `org.hibernate.tuple.entity.BytecodeEnhancementMetadataPojoImpl#hasUnFetchedAttributes` does, which is called through `hasUninitializedLazyProperties`